base_layer/core/src/chain_storage/lmdb_db/lmdb.rs

```diff
@@ -134,9 +134,17 @@ where
     }
 }
 
-pub fn lmdb_exists<K>(env: &Environment, db: &Database, key: &K) -> Result<bool, ChainStorageError>
-where K: Serialize {
+pub fn lmdb_get<K, V>(env: &Environment, db: &Database, key: &K) -> Result<Option<V>, ChainStorageError>
+where
+    K: Serialize,
+    V: DeserializeOwned,
+{
     let txn = ReadTransaction::new(env).map_err(|e| ChainStorageError::AccessError(e.to_string()))?;
+    lmdb_txn_get(&txn, db, key)
+}
+
+pub fn lmdb_txn_exists<K>(txn: &ConstTransaction<'_>, db: &Database, key: &K) -> Result<bool, ChainStorageError>
```

Comment:
I originally thought this meant "check if a transactions exists", so maybe `lmdb_exists_txn` better. It'll also show up in intellisense nearer the original method

Reply:
Ye ideally I think we should get rid of `lmdb_txn_exists` and just have `lmdb_exists` that takes a transaction (move the responsibility of creating db transactions in lmdb_db) - just didnt want to make too many changes

Reply:
But I agree with the name change - will resolve conflicts and rename